File: 2.0/java/contacts/getContactHistory.java
import com.callfire.api.client.CallfireClient;
import com.callfire.api.client.api.common.model.request.GetByIdRequest;
import com.callfire.api.client.api.contacts.model.ContactHistory;

class ApiClientSample {
    public static void main(String[] args) {
        CallfireClient client = new CallfireClient("api login", "api password");
        GetByIdRequest request = GetByIdRequest.create()
            .id(379506003L)
            .offset(0L)
            .limit(10L)
            .fields("calls/fromNumber,calls/toNumber,texts/message")
            .build();
        ContactHistory history = client.contactsApi().getHistory(request);
    }
}

File: 2.0/csharp/contacts/getContactHistory.cs
using CallfireApiClient;
using CallfireApiClient.Api.Common.Model.Request;
using CallfireApiClient.Api.Contacts.Model;

public class ApiClientSample
{
    public static void Main(string[] args)
    {
        var client = new CallfireClient("api_login", "api_password");
        var request = new GetByIdRequest
        {
            Id = 379506003,
            Offset = 0,
            Limit = 10,
            Fields = "calls/fromNumber,calls/toNumber,texts/message"
        };
        ContactHistory contactHistory = client.ContactsApi.GetHistory(request);
    }
}

